sw/device/tests/sram_ctrl_execution_test_main.c

```diff
@@ -0,0 +1,163 @@
+     *   31        25 24   20 19   15 14   12 11   7 6          0
+     *
+     *   Custom representation: add a5, a0, a5
+     *   +-----------+---------+---------+-------+--------+-----------+
+     *   |  0000000  |  01010  |  01011  |  000  | 01010  |  0110011  |
+     *   +-----------+---------+---------+-------+--------+-----------+
+     *   31        25 24     20 19     15 14   12 11     7 6          0
+     */
+    0x00A58533,
+    /*
+     *   Abstract representation: ret == jalr rd, offset(rs1)
+     *   +----------------+-------+-------+------+-----------+
+     *   |  000000000000  |  rs1  |  000  |  rd  |  1100111  |
+     *   +----------------+-------+-------+------+-----------+
+     *   31             20 19   15 14   12 11   7 6          0
+     *
+     *   Custom representation: jalr zero, 0(ra)
+     *   +----------------+---------+-------+---------+-----------+
+     *   |  000000000000  |  00001  |  000  |  00000  |  1100111  |
+     *   +----------------+---------+-------+---------+-----------+
+     *   31             20 19     15 14   12 11      7 6          0
+     */
+    0x00008067,
+};
+
+/*
+ * Adds the first parameter to the second parameter and returns the result.
+ *
+ * Maps to the `execution_test_instructions` buffer as the function body.
+ */
+typedef uint32_t (*ram_exec_function_t)(volatile uint32_t, volatile uint32_t);
+
+static bool otp_ifetch_enabled(void) {
+  otp_ctrl_t otp = otp_ctrl_init();
+
+  dif_otp_ctrl_config_t config = {
+      .check_timeout = 100000,
+      .integrity_period_mask = 0x3ffff,
+      .consistency_period_mask = 0x3ffffff,
+  };
+  CHECK_DIF_OK(dif_otp_ctrl_configure(&otp.dif, config));
+
+  CHECK_DIF_OK(dif_otp_ctrl_dai_read_start(&otp.dif, kDifOtpCtrlPartitionHwCfg,
+                                           kOtpIfetchHwRelativeOffset));
+
+  otp_wait_for_dai(&otp);
+
+  uint32_t value;
+  CHECK_DIF_OK(dif_otp_ctrl_dai_read32_end(&otp.dif, &value));
+
+  // OTP stores IFETCH state in a single bit (enabled/disabled).
+  return value & 0x00000001;
+}
+
+/*
+ * Performs SRAM execution test.
+ *
+ * `ram_exec_function_t` is "mapped" onto the `execution_test_instructions`
+ * buffer that holds hand-crafted machine instructions. Effectively, this
+ * buffer becomes the function body.
+ *
+ * According to the calling convention the parameters will be passed in the
+ * a0 and a1 registers, with a0 also treated as the return value register.
+ */
+static void sram_execution_test(void) {
+  // Map the function pointer onto the instruction buffer.
+  ram_exec_function_t func = (ram_exec_function_t)execution_test_instructions;
+
+  LOG_INFO("Executing from SRAM at = %x", func);
+  LOG_INFO("Instruction[0] = %x (add a0, a1, a0), at %x",
+           execution_test_instructions[0], &execution_test_instructions[0]);
+  LOG_INFO("Instruction[1] = %x (jalr ra, x0) at %x",
+           execution_test_instructions[1], &execution_test_instructions[1]);
```

Comment:
Maybe instead of logging these messages, you use the CHECK() macro instead (with the message as an error message if an unexpected instruction is loaded)? This would improve test runtime.

Reply:
> This would improve test runtime

Reply:
I thought only DV sim disables these by default. IIRC, Verilator sims (kicked off by pytest) use them by default (they seem to run on my machine by default anyway). @sriyerg can you confirm?

Reply:
I suppose you are talking about logging over UART? Then yes - it is enabled by default on Verilator and FPGA.

Reply:
I have deleted the debug output.